main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"log"
+	"regexp"
+	"strconv"
+	"sync"
+)
+
+func Serve(w http.ResponseWriter, r *http.Request) {
+	var h http.Handler
+	p := r.URL.Path
+	switch {
+		case match(p, "/"):
+			h = get(Home)
+		case match(p, "/postings"):
+			h = get(GetLatestPostings)
+		default:
+			http.NotFound(w, r)
+			return
+	}
+	h.ServeHTTP(w, r)
+}
+
+// match reports whether path matches ^regex$, and if it matches,
+// assigns any capture groups to the *string or *int vars.
+func match(path, pattern string, vars ...interface{}) bool {
+	regex := mustCompileCached(pattern)
+	matches := regex.FindStringSubmatch(path)
+	if len(matches) <= 0 {
```

Comment:
Return value of nil indicates no matches according to Go 1.17.5 docs.

snorlax@snorlax-bookworm:~$ go version
go version go1.17.5 linux/amd64

snorlax@snorlax-bookworm:~$ go doc regexp.Regexp.FindStringSubmatch
package regexp // import "regexp"

func (re *Regexp) FindStringSubmatch(s string) []string
    FindStringSubmatch returns a slice of strings holding the text of the
    leftmost match of the regular expression in s and the matches, if any, of
    its subexpressions, as defined by the 'Submatch' description in the package
    comment. A return value of nil indicates no match.